### PR TITLE
Migrate to storybook v6

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

Keeping up to date. Prepping for a future where docs are much more easily leveraged and are consumable within SB.

### WHAT is this pull request doing?

Upating to SB 6

We're stuck with knobs and contexts for the moment, as the modern
args/toolbars don't support a way of deeplinking to a specific setup
(e.g. "Show me this story with the new DL enabled")

### How to 🎩

Run SB and see that it works:

- AppProvider / Frame stories don't get double wrapped in an app provider
- Everything renders the same according to chromatic
- Toggling strict mode / new DL still works, and you can deep link to a specific story with the new DL enabled.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -29,49 +23,24 @@ module.exports = {
...
+                '../build/cache/storybook/markdown',
+              ),
             },
           },
           {
             loader: `${__dirname}/polaris-readme-loader.js`,
           },
         ],
       },
-      {
```
Comment (BPScott) [inline .storybook/main.js:49]: v6 as configures TS, and images to work out the box

File: .storybook/polaris-readme-loader.js [hunk 2]
```diff
@@ -44,13 +48,15 @@ export function ${example.storyName}() {
...
-  decorators: [withA11y],
-  parameters: {
-    notes: ${JSON.stringify(example.description)},
-    percy: {skip: ${JSON.stringify(!testIndividualExamples)}},
-  }
+
+${example.storyName}.storyName = ${JSON.stringify(example.name)};
+${example.storyName}.args = {omitAppProvider: ${readme.omitAppProvider}};
+${example.storyName}.parameters = {
+  layout: '${hasFullscreenLayout ? 'fullscreen' : 'padded'}',
```
Comment (BPScott) [inline .storybook/polaris-readme-loader.js:55]: The new layout parameter controls if a story renders flush to the edge ("full screen") or has a 1rem of padding ("padded"), which removes the need for our PaddingDecorator helper

File: .storybook/preview.js [hunk 3]
```diff
@@ -7,40 +7,25 @@ import DefaultThemeColors from '@shopify/polaris-tokens/dist-modern/theme/base.j
 import {AppProvider} from '../src';
 import enTranslations from '../locales/en.json';
 
-addParameters({
-  options: {
-    showRoots: true,
```
Comment (BPScott) [inline .storybook/preview.js:12]: This is now true by default so no need to configure it

File: package.json [hunk 4]
```diff
@@ -110,30 +110,27 @@
     "@shopify/sewing-kit": "^0.132.2",
     "@shopify/shrink-ray": "^2.3.1",
     "@shopify/splash": "^0.0.8",
-    "@storybook/addon-a11y": "^5.3.19",
-    "@storybook/addon-actions": "^5.3.19",
+    "@storybook/addon-a11y": "^6.0.12",
     "@storybook/addon-console": "^1.2.1",
     "@storybook/addon-contexts": "^5.3.19",
```
Comment (BPScott) [inline package.json:115]: Contexts is deprecated and doesn't have a v6 version, but we're stuck with it as the new args/toolbars approach of configuring the "Strict Mode" and "Design Language" toggle buttons doesn't support deeplinking. I've raised a bug to request support for that so we can move away: https://github.com/storybookjs/storybook/issues/11604

**Comment by kaelig** [comment]
In the "Docs" tab, "All examples" stories are adding noise at the top of the page. What do you think about removing them from the docs tab? ([this section of Storybook's documentation](https://storybook.js.org/docs/html/writing-docs/docs-page#story-level) make me think it's possible to do so)

**Comment by BPScott** [comment]
Found an option for that by spelunking the source code, because it wasn't documented :)

**Comment by kaelig** [comment]
Ha! Good find 🕵️

**Comment by kaelig** [comment]
Can you also update the changelog?

**Pull request merged**